compiler/rustc_macros/src/type_foldable.rs

```diff
@@ -44,14 +44,51 @@ pub(super) fn type_foldable_derive(mut s: synstructure::Structure<'_>) -> proc_m
         })
     });
 
+    let body_fold = s.each_variant(|vi| {
+        let bindings = vi.bindings();
+        vi.construct(|_, index| {
+            let bind = &bindings[index];
+
+            let mut fixed = false;
+
+            // retain value of fields with #[type_foldable(identity)]
+            bind.ast().attrs.iter().for_each(|x| {
+                if !x.path().is_ident("type_foldable") {
+                    return;
+                }
+                let _ = x.parse_nested_meta(|nested| {
+                    if nested.path.is_ident("identity") {
+                        fixed = true;
+                    }
+                    Ok(())
+                });
+            });
```

Comment:
move this into a shared closure?

Reply:
or also add a `has_ignore_attr` function to `rustc_macros`, similar to `rustc_type_ir_macros`